yarl/_url.py

```diff
@@ -255,7 +273,7 @@ def __new__(
                 username, password, host, port = cls._split_netloc(val[1])
                 if host is None:
                     raise ValueError("Invalid URL: host is required for absolute urls")
-                host = cls._encode_host(host)
+                host = cls._encode_host(host, validate_host=False)
```

Comment:
We split / parsed the url so it never going to have the same concerns as #880 so I turned it off here.  There is different validation in urllib which raises different errors so I added a test for that. We may need to do a followup to smooth out this rough edge, but since the case in #880 is addressed, I figure this is good enough for a first pass.